templates/debug_panel.html.twig

```diff
@@ -0,0 +1,188 @@
+{#
+ # ---------------------------------------------------------------------
+ #
+ # GLPI - Gestionnaire Libre de Parc Informatique
+ #
+ # http://glpi-project.org
+ #
+ # @copyright 2015-2022 Teclib' and contributors.
+ # @copyright 2003-2014 by the INDEPNET Development Team.
+ # @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ #
+ # ---------------------------------------------------------------------
+ #
+ # LICENSE
+ #
+ # This file is part of GLPI.
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # ---------------------------------------------------------------------
+ #}
+
+{% set rand = rand|default(random()) %}
+
+{% macro print_clean_array(data, pad = 0, js_expand = false) %}
+   {% if data|length > 0 %}
+      <table class="array-debug table table-striped card-table">
+         <tr>
+            <th>KEY</th>
+            <th>=></th>
+            <th>VALUE</th>
+         </tr>
+         {% for key, value in data %}
+            {% set row_rand = random() %}
+            <tr>
+               <td>{{ key }}</td>
+               <td>
+                  {% if js_expand and value is array %}
+                     <a class="fw-bolder" href="javascript:showHideDiv('content{{ key ~ row_rand }}', '', '', '')">=></a>
+                  {% else %}
+                     =>
+                  {% endif %}
+               </td>
+               <td>
+                  {% if value is array %}
+                     <div id="content{{ key ~ row_rand }}" {{ js_expand ? 'style="display: none"' : '' }}>
+                        {{ _self.print_clean_array(value, pad + 1) }}
+                     </div>
+                  {% else %}
+                     {% if value is same as(true) or value is same as(false) %}
+                        {{ value ? 'true' : 'false' }}
+                     {% elseif value is object %}
+                        {{ call([value, '__toString'])|default('(object) ' ~ get_class(value)) }}
+                     {% else %}
+                        {{ value }}
```

Comment:
I'm not sure that adding something that displays modified values is a good idea. It could be confusing if the slashes were present in the original value or if they were added by json_encode.
Using `dump()` to display the values shows type information without modifying string values beyond surrounding them with quotes. It also enables extra functionality like being able to view object data beyond just the class name.

Reply:
Indeed, `dump` is a better alternative.

On objects dumps, I have encoding issues :
![image](https://user-images.githubusercontent.com/33253653/207252239-b8444f3c-097f-4ff2-bbf4-904f02cd75a7.png)

Reply:
I don't have this issue

Reply:
I do not have `xdebug` loaded and I have `html_errors=0` in my PHP config. Maybe it is due to this.

https://github.com/twigphp/Twig/blob/b5ec7ab1cdfebf3e0ab0a5fddc342dbd903763fa/src/Extension/DebugExtension.php#L20-L27

Reply:
Problem was due to the following syntax, that was resulting in having content escaped when printed by `{{ dump(value) }}` and then escaped again when printed by `{{ out[1]|trim }}`.
```
{% set out %}
   {{ dump(value) }}
{% endset %}
```
Using a `{% set out = dump(value) %}` prevents the first escaping.